lib/github-pages-health-check.rb

```diff
@@ -65,11 +66,15 @@ def valid_domain?
       PublicSuffix.valid? domain
     end
 
-    # Is this domain an SLD, meaning a CNAME would be innapropriate
+    # Is this domain an apex domain, meaning a CNAME would be innapropriate
     def apex_domain?
-      PublicSuffix.parse(domain).trd == nil
-    rescue
-      false
+      return @apex_domain if defined?(@apex_domain)
+
+      answers = Resolv::DNS.open { |dns|
+        dns.getresources(absolute_domain, Resolv::DNS::Resource::IN::NS)
```

Comment:
Unfortunately the DNS lookup we're already doing doesn't fetch NS records, it just asks for an A record. I don't think there's a way to look up multiple record types (except an ANY request - [but that's on its way out](https://blog.cloudflare.com/deprecating-dns-any-meta-query-type/))